src/middleware/basicAuth.js

```diff
@@ -27,11 +31,24 @@ const basicAuthMiddleware = function (request, response, callback) {
         .toString('utf8')
         .split(':');
 
-    if (username === config.basicAuthUser.username && password === config.basicAuthUser.password) {
+    if (!PER_USER_BASIC_AUTH && username === config.basicAuthUser.username && password === config.basicAuthUser.password) {
```

Comment:
I think you've covered this right?

Reply:
I was actually thinking to not use the default basic auth at all while `PER_USER_BASIC_AUTH` is on. The setting I think should be activated once basic setup is achieved. If no users are valid then no login should be possible until `PER_USER_BASIC_AUTH` is turned off again. This its safer to deny rather than fall back

Reply:
With `PER_USER_BASIC_AUTH` and `!enableUserAccounts` I was thinking it would default to `default-user:` for the basic

Reply:
How about we also remove the

```js
else if (user && user.enabled && !user.password && !password) {
    // Login to an account without password
    return callback();
}
```

That way `default-user` cannot be used with basic at all, as well as any other no password account

Reply:
Also cleaned up the code so that the basic logic in user.js matches the login in basicAuth